### PR TITLE
Improve callbacks

### PR DESCRIPTION
**Pull request by ethanfrey**
Callbacks always go to original caller.
They can specify a clear ID to receive it.

TODO:

- [ ] Add ts-level tests that callbacks work (see #14)

### CODE REVIEW

File: contracts/simple-ica-controller/src/contract.rs [hunk 1]
```diff
@@ -32,15 +32,15 @@ pub fn execute(deps: DepsMut, env: Env, info: MessageInfo, msg: ExecuteMsg) -> S
...
+            callback_id: callback,
         } => execute_send_msgs(deps, env, info, channel_id, msgs, callback),
         ExecuteMsg::CheckRemoteBalance { channel_id } => {
             execute_check_remote_balance(deps, env, info, channel_id)
         }
         ExecuteMsg::IbcQuery {
             channel_id,
             msgs,
-            callback,
+            callback_id: callback,
```
Comment (webmaster128) [inline contracts/simple-ica-controller/src/contract.rs:43]: Can you rename the local variables `callback` -> `callback_id` too?

File: contracts/simple-ica-controller/src/ibc.rs [hunk 2]
```diff
@@ -119,21 +125,20 @@ fn acknowledge_dispatch(
...
             let msg = WasmMsg::Execute {
-                contract_addr: callback.clone(),
-                msg: to_binary(&ReceiveIbcResponseMsg { msg })?,
+                contract_addr: sender,
+                msg: to_binary(&ReceiveIbcResponseMsg { id, msg })?,
                 funds: vec![],
             };
             Ok(IbcBasicResponse::new()
                 .add_attribute("action", "acknowledge_dispatch")
-                .add_attribute("callback_address", callback)
```
Comment (webmaster128) [inline contracts/simple-ica-controller/src/ibc.rs:136]: We could add callback_id to the event? Not sure if this is useful. Maybe good for debugging.

File: packages/simple-ica/src/ibc_msg.rs [hunk 3]
```diff
@@ -70,6 +72,7 @@ impl StdAck {
 #[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
 #[serde(rename_all = "snake_case")]
 pub struct ReceiveIbcResponseMsg {
+    pub id: String,
```
Comment (webmaster128) [inline packages/simple-ica/src/ibc_msg.rs:75]:
```suggestion
    /// The ID choosen by the caller in the `callback_id`
    pub id: String,
```

**Comment by ethanfrey** [comment]
@webmaster128 I resolved the comments.

This has no tests on callback, but neither did the previous version. Maybe we can merge this while I get #14 finished and use that to ensure callbacks work properly?

**Comment by JakeHartnell** [comment]
> @webmaster128 Thanks for the review. I resolved the comments.
> 
> This has no tests on callback, but neither did the previous version. Maybe we can merge this while I get #14 finished and use that to ensure callbacks work properly?

Can make another PR for more tests there.

**Comment by ethanfrey** [comment]
@JakeHartnell #14 had the callback tests and fixes two bugs in the rust code in addition to a sample contract to capture callbacks.

Happy for your review on that one

**Pull request merged**